src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImpl.java

```diff
@@ -90,16 +89,88 @@ void postConstruct() {
         }
     }
 
+    @Override
+    public BinaryData getUnstructuredAudioBlob(UUID location) {
+        return dataManagementApi.getBlobDataFromUnstructuredContainer(location);
+    }
+
+    @Override
+    public BinaryData getOutboundAudioBlob(UUID location) {
+        return dataManagementApi.getBlobDataFromOutboundContainer(location);
+    }
+
+    @Override
+    public UUID saveAudioBlobData(BinaryData binaryData) {
+        return dataManagementApi.saveBlobDataToOutboundContainer(binaryData);
+    }
+
+    @Override
+    public TransientObjectDirectoryEntity saveTransientDataLocation(MediaRequestEntity mediaRequest,
+                                                                    UUID externalLocation) {
+        return transientObjectDirectoryService.saveTransientDataLocation(mediaRequest, externalLocation);
+    }
+
+    @Override
+    public List<MediaEntity> getMediaMetadata(Integer hearingId) {
+        return mediaRepository.findAllByHearingId(hearingId);
+    }
+
+    @Override
+    public Optional<UUID> getMediaLocation(MediaEntity media) {
+        Optional<UUID> externalLocation = Optional.empty();
+
+        ObjectDirectoryStatusEntity objectDirectoryStatus = objectDirectoryStatusRepository.getReferenceById(STORED.getId());
+        ExternalLocationTypeEntity externalLocationType = externalLocationTypeRepository.getReferenceById(UNSTRUCTURED.getId());
+        List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntityList = externalObjectDirectoryRepository.findByMediaStatusAndType(
+            media, objectDirectoryStatus, externalLocationType
+        );
+
+        if (!externalObjectDirectoryEntityList.isEmpty()) {
+            if (externalObjectDirectoryEntityList.size() != 1) {
+                log.warn(
+                    "Only one External Object Directory expected, but found {} for mediaId={}, statusEnum={}, externalLocationTypeId={}",
+                    externalObjectDirectoryEntityList.size(),
+                    media.getId(),
+                    STORED,
+                    externalLocationType.getId()
+                );
+            }
+            externalLocation = Optional.ofNullable(externalObjectDirectoryEntityList.get(0).getExternalLocation());
+        }
+
+        return externalLocation;
+    }
+
+    @Override
+    public Path saveBlobDataToTempWorkspace(BinaryData mediaFile, String fileName) throws IOException {
+
+        return fileOperationService.saveFileToTempWorkspace(mediaFile, fileName);
+    }
+
+    @Override
+    @Transactional
+    public UUID saveProcessedData(MediaRequestEntity mediaRequest, BinaryData binaryData) {
+        UUID blobId = saveAudioBlobData(binaryData);
+        saveTransientDataLocation(mediaRequest, blobId);
+
+        return blobId;
+    }
+
+    @Override
+    public void handleKedaInvocationForMediaRequests() {
```

Comment:
This top level service looks like it's missing `@Transactional` for all the various saves nested in `processAudioRequest`?

Reply:
I think we explictly want to avoid `@Transactional` for this particular case - if there is an issue/exception thrown during processing we don't want to role-back any changes to `media_request.request_status`. The approach in `processAudioRequest()` is to catch any exception and explictly set state to FAILED.